### PR TITLE
IDEX-3805 Migrate to HttpJsonRequestFactory

### PR DESCRIPTION
**Pull request by skabashnyuk**

### CODE REVIEW

File: platform-api/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/RemoteOAuthTokenProvider.java [hunk 1]
```diff
@@ -35,18 +35,27 @@
...
+            HttpJsonRequestFactory httpJsonRequestFactory) {
+        this.apiEndpoint = apiEndpoint;
+        this.httpJsonRequestFactory = httpJsonRequestFactory;
+    }
 
     /** {@inheritDoc} */
     @Override
     public OAuthToken getToken(String oauthProviderName, String userId) throws IOException {
-        if (userId.isEmpty()) {
+        if (userId == null || userId.isEmpty()) {
```
Comment (sleshchenko) [inline platform-api/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/RemoteOAuthTokenProvider.java]: Strings.isNullOrEmtpty?

**Comment by mshaposhnik** [comment]
:fire:

**Comment by garagatyi** [comment]
ok

**Comment by sleshchenko** [comment]
nice

**Pull request merged**